portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -160,12 +182,16 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     searchQuery = searchQuery.substring(1,searchQuery.length() - 1);
     int pageToken = Integer.parseInt(
       (String) request.getParameter(RequestParameters.PAGE_TOKEN));
+    String languageCode = (String) request.getParameter(
+      RequestParameters.LANGUAGE); 
+    languageCode = languageCode.substring(1,languageCode.length() - 1);
 
     PreparedQuery results = getAllComments();
     ArrayList<Comment> unfilteredComments = new ArrayList<Comment>();
-    results.asIterable().forEach(commentEntity -> {
-      unfilteredComments.add(Comment.fromEntity(commentEntity));
-    });
+    for (Entity commentEntity : results.asIterable()) {
```

Comment:
this seems like it could still use the forEach. 
Could you explain the pros of switching?

Reply:
I was getting a compiler error saying something about not being able to modify local variables within a lambda, and I realized it was because translateComment() modifies the comment object. I don't know why this is a rule, but when I searched StackOverflow for the issue people were saying just to use  for-each loop. I considered ways to make it so the comment object wasn't modified (like change fromEntity to automatically translate the comment, or making the translateComment() method return a new comment object, or creating another function that simply contained the line        `unfilteredComments.add(Comment.fromEntity(commentEntity).translateComment(languageCode));` to pass to the forEach() function so that the local variable modification was happening somewhere else) but I was worried it would hurt my code readability/space efficiency. 

Do you know of any ways around this "no modifying local variables inside a lambda" rule? I'd really appreciate the help because I've gotten this error a few times now and it's very confusing to debug.